cpp/src/io/parquet/writer_impl.cu

```diff
@@ -384,15 +384,18 @@ writer::impl::impl(std::unique_ptr<data_sink> sink, writer_options const &option
   stats_granularity_(options.stats_granularity),
   out_sink_(std::move(sink)){}
 
-void writer::impl::write(table_view const &table, const table_metadata *metadata, cudaStream_t stream) {
+std::unique_ptr<std::vector<uint8_t>> writer::impl::write(
+                         table_view const &table, const table_metadata *metadata,
+                         bool return_filemetadata, const std::string& metadata_out_file_path,
+                         cudaStream_t stream) {    
   pq_chunked_state state;
   state.user_metadata = metadata;
   state.stream = stream;
   state.single_write_mode = true;
```

Comment:
Seems like `pq_chunked_state` is just a `struct` ...
```cpp
   struct pq_chunked_state {
      /// The writer to be used
      std::unique_ptr<writer>             wp;  
      /// Cuda stream to be used
      // ...
```
Can we add a constructor? So we don't have to initialize all these member after the default ctor is called?

Reply:
I don't think so, as this is intended to be a somewhat opaque shared_ptr blob in its primary use (declared locally here to share the same code path between chunked and non-chunked writes). Really out of the scope of this PR, though.

Reply:
That's fair. I set up an issue to address in a different PR later. https://github.com/rapidsai/cudf/issues/4735